pkg/utils/resources/suite_test.go

```diff
@@ -0,0 +1,393 @@
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/aws/karpenter-core/pkg/test"
+)
+
+func TestResources(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Resources Suite")
+}
+
+var _ = Describe("Resources Handling", func() {
+	Context("Resource Calculations", func() {
+
+		It("should schedule based on the sum of containers and sidecarContainers", func() {
+			pod := test.Pod(test.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{
+					Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("2"), v1.ResourceMemory: resource.MustParse("1Gi")},
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("2")},
+				},
+
+				InitContainers: []v1.Container{
+					{
+						RestartPolicy: lo.ToPtr(v1.ContainerRestartPolicyAlways),
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("4"), v1.ResourceMemory: resource.MustParse("2Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1")},
+						},
+					},
+				},
+			})
+			podResources := Ceiling(pod)
+			Expect(podResources.Limits.Cpu().CmpInt64(6)).To(Equal(0))
+			Expect(podResources.Limits.Memory().CmpInt64(3221225472)).To(Equal(0))
+		})
+
+		It("should schedule based on the sum of containers , sidecarContainers and overhead with max of initcontainers", func() {
+			pod := test.Pod(test.PodOptions{
+				Overhead: v1.ResourceList{
+					v1.ResourceCPU:    resource.MustParse("5"),
+					v1.ResourceMemory: resource.MustParse("500Mi"),
+				},
+				ResourceRequirements: v1.ResourceRequirements{
+					Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("2"), v1.ResourceMemory: resource.MustParse("1Gi")},
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("2")},
+				},
+
+				InitContainers: []v1.Container{
+					{
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("4"), v1.ResourceMemory: resource.MustParse("2Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1")},
+						},
+					},
+					{
+						RestartPolicy: lo.ToPtr(v1.ContainerRestartPolicyAlways),
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("3"), v1.ResourceMemory: resource.MustParse("3Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1")},
+						},
+					},
+				},
+			})
+			podResources := Ceiling(pod)
+			Expect(podResources.Limits.Cpu().CmpInt64(10)).To(Equal(0))
+			Expect(podResources.Limits.Memory().CmpInt64(4819255296)).To(Equal(0))
+		})
+
+		It("should schedule based on the max resource requests of containers and initContainers with sidecarContainers for order of type1", func() {
```

Comment:
```suggestion
		It("should schedule based on the max resource requests of containers and initContainers with sidecarContainers (native sidecar container comes first)", func() {
```

Mainly a comment on being more descriptive with what you mean by "types" and putting it into the test description so that someone can glance over this quickly and understand _immediately_ what it's doing without have to read the test details